File: 20260505/exos/02_condition.py
# ============================================================
# Resident Evil — statut médical
# ============================================================
# Dans le labo Umbrella, chaque survivant a un écran médical.
# L'écran affiche "En vie" si les PV sont strictement supérieurs à 0,
# "Mort" sinon. Leon vient d'être touché — vérifie son statut.
#
# Lance : python 02_condition.py
# ============================================================

pv = 18

# Détermine le statut selon les PV.
# Si pv est strictement supérieur à 0, statut vaut "En vie".
# Sinon (pv vaut 0 ou moins), statut vaut "Mort".
#
# pv = 18  →  18 > 0 ? oui  →  statut = "En vie"
# pv = 0   →  0 > 0  ? non  →  statut = "Mort"
#
# Résultat attendu : statut == "En vie"  (car pv vaut 18 ici)
#
# Indice : if / else

if pv > 0:
    statut = "En vie"
else:
    statut = "Mort"


# --- Vérification (ne pas modifier) ---
assert statut == "En vie", (
    f"Obtenu : '{statut}' — avec pv > 0, le statut attendu est « En vie »."
)
print("✅ Correct !")

File: 20260717/exos/02_condition.lua
-- ============================================================
-- Resident Evil — état de santé
-- ============================================================
-- L'écran médical de Leon change selon ses PV :
-- "Critique" en dessous de 30, "Blessé" entre 30 et 69,
-- "Stable" à 70 ou plus.
--
-- Lance : lua 02_condition.lua
-- ============================================================

vie = 45
etat = ""

-- Remplis etat selon la valeur de vie.
--
-- vie = 20  →  20 < 30 ? oui           →  etat = "Critique"
-- vie = 45  →  45 < 30 ? non, 45 < 70 ? oui  →  etat = "Blessé"
-- vie = 80  →  80 < 30 ? non, 80 < 70 ? non  →  etat = "Stable"
--
-- Résultat attendu : etat == "Blessé"  (car vie vaut 45 ici)
--
-- Indice : if / elseif / else / end

-- À toi :
if vie < 30 and vie > 0 then
    etat = "Critique"
elseif vie >= 30 and vie < 70 then
    etat = "Blessé"
else
    etat = "Stable"

end
 

-- --- Vérification (ne pas modifier) ---
assert(etat == "Blessé", "Obtenu : '" .. etat .. "' — avec vie = 45, attendu : Blessé")
print("✅ Correct !")
